pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,273 @@
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %#v: %v", node, err))
+		return
+	}
+	ctrl.nodeQueue.Add(key)
+}
+
+func (ctrl *Controller) updateNode(old, cur interface{}) {
+	oldNode := old.(*osev1.Node)
+	newNode := cur.(*osev1.Node)
+	if !reflect.DeepEqual(oldNode.Spec, newNode.Spec) {
+		// skipping the update in case of the Worker-Latency-Profile type transition from "Default" to "LowUpdateSlowReaction" and vice-versa
+		// (TODO) Ideally the user request has to be honoured, the transition need to be from Default -> Medium -> Low or vice-versa.
+		// Restricting the request for now until this process is automated in future.
+		if (oldNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction && newNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction) || (oldNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction && newNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction) {
+			glog.Infof("Skipping the Update Node event, name: %s, transition not allowed from old WorkerLatencyProfile: %s to new WorkerLatencyProfile: %s", newNode.Name, oldNode.Spec.WorkerLatencyProfile, newNode.Spec.WorkerLatencyProfile)
+			return
+		}
+		glog.V(4).Infof("Update Node event, name: %s", newNode.Name)
+		ctrl.enqueueNode(newNode)
+	}
+}
+
+func (ctrl *Controller) addNode(obj interface{}) {
+	node := obj.(*osev1.Node)
+	glog.V(4).Infof("Add Node event %s", node.Name)
+	ctrl.enqueueNode(node)
+}
+
+func (ctrl *Controller) deleteNode(obj interface{}) {
+	node, ok := obj.(*osev1.Node)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			return
+		}
+		node, ok = tombstone.Obj.(*osev1.Node)
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a KubeletConfig %#v", obj))
+			return
+		}
+	}
+	glog.V(4).Infof("Deleted Node %s and restored default config", node.Name)
+}
+
+// isStatusReady returns a boolean based on the condition statuses present in the Node CR status.
+// If the status condition type is Progressing or Degraded, this function returns false
+func isStatusReady(node *osev1.Node) bool {
+	if node == nil {
+		return true
+	}
+	if len(node.Status.WorkerLatencyProfileStatus.Conditions) > 0 {
+		for _, condition := range node.Status.WorkerLatencyProfileStatus.Conditions {
+			if condition.Type == osev1.WorkerLatencyProfileProgressing || condition.Type == osev1.WorkerLatencyProfileDegraded {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// updateNodestatus updates the status of the node object based on the machineconfigpool status
+func (ctrl *Controller) updateNodestatus(kc *mcfgv1.KubeletConfig, node *osev1.Node) error {
+	if kc == nil || node == nil {
+		return fmt.Errorf("unable to update the node status, incomplete data, kubelet config: %v, node: %v", kc, node)
```

Comment:
what does node = nil mean in this case? why call updateNodestatus if node is nil in the first place?

Reply:
Just to avoid any unwanted panics in case of accessing a nil object, this check is added.